### PR TITLE
Xiaoxu/fixcircula

### PR DESCRIPTION
**Pull request by xiaoxuqi-ms**

### CODE REVIEW

File: test/roundtripValidatorTests.ts [hunk 1]
```diff
@@ -7,6 +7,31 @@ jest.setTimeout(999999);
 
 describe.skip("Live Validator", () => {
```
Comment (leniatgh) [inline test/roundtripValidatorTests.ts]: the test case has been skipped in my last commit. Please revert it

File: lib/swagger/jsonLoader.ts [hunk 2]
```diff
@@ -111,6 +111,65 @@ export class JsonLoader implements Loader<Json> {
...
+        fileContent,
+        resolveOption
+      );
+      spec = await parser.dereference(
+          this.fileLoader.resolvePath(filePath),
+        spec,
+        resolveOption
+      );
+      removeProperty(spec);
+        return spec;
```
Comment (leniatgh) [inline lib/swagger/jsonLoader.ts]: need format code

File: lib/swagger/jsonLoader.ts [hunk 3]
```diff
@@ -111,6 +111,65 @@ export class JsonLoader implements Loader<Json> {
     // throw new Error(`Unknown file format while loading file ${cache.filePath}`);
   }
 
+  private getFileContent(filePath: string, fileString: string): any {
+    if (
+      this.opts.supportYaml &&
```
Comment (leniatgh) [inline lib/swagger/jsonLoader.ts]: why yaml support is implemented in this pr?

File: lib/liveValidation/liveValidatorLoader.ts [hunk 4]
```diff
@@ -124,8 +124,16 @@ export class LiveValidatorLoader implements Loader<SwaggerSpec> {
...
   }
 
+  public resetShouldResolveRef() {
+    this.jsonLoader.resetShouldResolveRef();
+    this.opts.shouldResolveRef = false;
+  }
+
   public async load(specFilePath: string): Promise<SwaggerSpec> {
     const spec = await this.swaggerLoader.load(specFilePath);
+    if (this.opts.shouldResolveRef === true) {
```
Comment (leniatgh) [inline lib/liveValidation/liveValidatorLoader.ts]:
```suggestion
    if (this.opts.shouldResolveRef) {
```

File: lib/swagger/jsonLoader.ts (hunk 3 above)
Comment (xiaoxuqi-ms) [reply lib/swagger/jsonLoader.ts]: can be removed.

**Comment by raych1** [comment]
This is invalid PR.

**Pull request closed without merging**